### PR TITLE
Update displaylink version to 2.3b for yosemite + change license to gratis

### PR DESCRIPTION
**Pull request by donbobka**
- Update displaylink version to 2.3b for yosemite
- Change license to commercial

### CODE REVIEW

File: Casks/displaylink.rb [hunk 1]
```diff
@@ -1,16 +1,21 @@
 cask :v1 => 'displaylink' do
-  version '2.2'
-  sha256 '5c9a97a476b5ff27811491eebb653a03c96f899562b67566c24100d8593b1daa'
+  if MacOS.release < :yosemite
```
Comment (vitorgalvao) [inline Casks/displaylink.rb]: We prefer this to be inclusive (i.e. `<= :maverick`).

**Comment by donbobka** [comment]
Fixed
+ I added version 1.7 for tiger and leopard but i can't check it

File: Casks/displaylink.rb [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
+    version '1.7'
+    sha256 'b35dc49fe286aa858d7f6f44fd3f6703de83fae2316d20b05e637a1134ba2440'
+  elsif MacOS.release <= :snow_leopard
+    version '2.2'
+    sha256 '5c9a97a476b5ff27811491eebb653a03c96f899562b67566c24100d8593b1daa'
+  elsif MacOS.release <= :yosemite
+    version '2.3b'
+    sha256 'abcfd2355bb6c5acb19d7ae143c058560fd244330217015fea978c5426f261d3'
+  else
+    # Unsupported OSX version: Please update cask and check it.
```
Comment (vitorgalvao) [inline Casks/displaylink.rb]: The cask is breaking on account of this being empty. Not your fault, though, and nothing you can do yet. It’s a similar case to https://github.com/caskroom/homebrew-versions/pull/590. Please change the comment, though (we have a standardised one) to

```
# Unusual case: there is no fall-through.  Each version of the software is
# specific to an OS X release, so define nothing when the release is unknown.
```

And please [squash your commits](http://davidwalsh.name/squash-commits-git). After #5248 is done, we can merge this in.

**Comment by donbobka** [comment]
Ready

**Comment by donbobka** [comment]
Change license to gratis

File: Casks/displaylink.rb [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
+    version '1.7'
+    sha256 'b35dc49fe286aa858d7f6f44fd3f6703de83fae2316d20b05e637a1134ba2440'
+  elsif MacOS.release <= :snow_leopard
+    version '2.2'
+    sha256 '5c9a97a476b5ff27811491eebb653a03c96f899562b67566c24100d8593b1daa'
+  elsif MacOS.release <= :yosemite
+    version '2.3b'
+    sha256 'abcfd2355bb6c5acb19d7ae143c058560fd244330217015fea978c5426f261d3'
+  else
+    # Unusual case: there is no fall-through.  Each version of the software is
```
Comment (vitorgalvao) [inline Casks/displaylink.rb]: I’ll ask for a new change to the comment. It appears [we’re changing the template](https://github.com/caskroom/homebrew-cask/pull/8657/files).

```
# Unusual case: there is no fall-through.  The software will stop
# working, or is dangerous to run, on the next OS X release.
```

**Comment by donbobka** [comment]
Fixed

**Comment by vitorgalvao** [comment]
I guess you didn’t saw [the edit](https://github.com/caskroom/homebrew-cask/pull/8624#discussion_r22470910).

Looking at it closer, though, is `elsif MacOS.release <= :yosemite` really needed? Couldn’t it probably be just an `else`? I see nothing to suggest this changes with every OS X version, and that on the new one the app will break.

**Comment by donbobka** [comment]
My opinion, you are right. Just `else` will be more adequate. In case if next osx release will break something, somebody will make a pull request to repair cask.

**Comment by donbobka** [comment]
All casks with `url :unknown` will be broken right after new osx release. It's mean that maintainers should update a lot of casks at once. Option `url :unknown` should be exceptional.

**Comment by donbobka** [comment]
@vitorgalvao, I updated cask. 
If you agree, i will squash commits.

**Comment by vitorgalvao** [comment]
Looks good. Please squash.

**Comment by donbobka** [comment]
Ready.
Why uninstall section has a lot of commented code? May be delete it?

**Comment by vitorgalvao** [comment]
Actually, I’ve just tested it and it seems that much of it is very much necessary.

```
  uninstall :pkgutil => [
                         'com.displaylink.displaylinkdriversigned',
                         'com.displaylink.displaylinkdriverunsigned'
                        ],
            # :kext => 'com.displaylink.driver.DisplayLinkDriver', # causes kernel panic
            :launchctl => [
                           'com.displaylink.displaylinkmanager',
                           'com.displaylink.useragent'
                          ],
            :quit => 'DisplayLink.DisplayLinkUserAgent',
            :delete => '/Applications/DisplayLink'
```

**Comment by donbobka** [comment]
A little bit later i can check uninstaller with connected device.

**Comment by vitorgalvao** [comment]
@donbobka Any progress?

**Comment by donbobka** [comment]
- Update to version 2.3
- remove uninstall stanza, because it doesn't work.

**Comment by vitorgalvao** [comment]
We should fix it instead, not remove it. I’ll take a look into it later.

**Comment by donbobka** [comment]
The problem in following line: `:kext => 'com.displaylink.driver.DisplayLinkDriver', # causes kernel panic`.
Uninstaller doesn't remove driver without this line.

**Comment by vitorgalvao** [comment]
But it should remove other resources, which is already something. That line would be a warning that that specific issue needs to be solved. We have to check why it happens an fix it.

**Comment by donbobka** [comment]
With uninstall stanza from 3f264b24d99fcae90de85df847159ce05dc0c821 
```
donbobka-air:~ donbobka$ brew cask uninstall  displaylink
==> Running uninstall process for displaylink; your password may be necessary
==> Removing launchctl service com.displaylink.useragent-prelogin
Password:
==> Removing launchctl service com.displaylink.useragent
==> /Users/donbobka/com.displaylink.useragent: No such file or directory
==> Removing launchctl service com.displaylink.displaylinkmanager
==> /Users/donbobka/com.displaylink.displaylinkmanager: No such file or director
==> Quitting application ID DisplayLinkUserAgent
==> Removing files from pkgutil Bill-of-Materials
==> Removing files: ["/Applications/DisplayLink"]

donbobka-air:~ donbobka$ ls /Library/Launch*/*display*
/Library/LaunchAgents/com.displaylink.useragent-prelogin.plist
/Library/LaunchAgents/com.displaylink.useragent.plist
/Library/LaunchDaemons/com.displaylink.displaylinkmanager.plist
```
Why it searches in current directory? How I can remove following plists: `/Library/LaunchAgents/com.displaylink.useragent-prelogin.plist`, `/Library/LaunchAgents/com.displaylink.useragent.plist`, `/Library/LaunchDaemons/com.displaylink.displaylinkmanager.plist`?

**Comment by vitorgalvao** [comment]
You can remove them with `:delete`. `:launchctl` just quits the processes.

**Comment by donbobka** [comment]
Ok, now only problem is a kext.
```
kextstat | grep display
  125    0 0xffffff7f829e6000 0xe000     0xe000     com.displaylink.driver.DisplayLinkDriver (2.3) <75 5 4 3>
```

**Comment by vitorgalvao** [comment]
It is. At the time I’ve tried a few more casks, and this was the only one that caused a kernel panic, so it does seems to be something with the app itself, and not homebrew-cask. Unless you have an idea for a solution, maybe we should leave it there commented for now and take a look at another time.

**Comment by donbobka** [comment]
Unfortunately, have no idea. Just rebased all my commits

**Comment by vitorgalvao** [comment]
We’ll have a look later, then. Thank you.

**Pull request merged**